### ISSUE TITLE
[Bug] Overlay stays after dialog window is closed

### ISSUE THREAD

**Issue by mts527** [magento/magento2-phpstorm-plugin#667]
**Describe the bug** (*)

When creating a new Magento 2 module an overlay stays over the editor which requires PhpStorm to restart so the overlay is cleared and editor usable again.

```
com.intellij.util.indexing.impl.MapReduceIndexMappingException: java.lang.NullPointerException
	at com.intellij.util.indexing.impl.MapReduceIndex.mapInputAndPrepareUpdate(MapReduceIndex.java:272)
	at com.intellij.indexing.composite.CompositeInvertedIndexBase.updateBaseIndex(CompositeInvertedIndexBase.java:232)
	at com.intellij.indexing.composite.CompositeInvertedIndexBase.mapInputAndPrepareUpdate(CompositeInvertedIndexBase.java:55)
	at com.intellij.indexing.composite.CompositeInvertedIndexBase.mapInputAndPrepareUpdate(CompositeInvertedIndexBase.java:26)
	at com.intellij.util.indexing.FileBasedIndexImpl.lambda$indexUnsavedDocument$14(FileBasedIndexImpl.java:1081)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.freezeFileTypeTemporarilyIn(FileTypeManagerImpl.java:644)
	at com.intellij.util.indexing.FileBasedIndexImpl.indexUnsavedDocument(FileBasedIndexImpl.java:1052)
	at com.intellij.util.indexing.RegisteredIndexes$DocumentUpdateTask.doProcess(RegisteredIndexes.java:177)
	at com.intellij.util.indexing.RegisteredIndexes$DocumentUpdateTask.doProcess(RegisteredIndexes.java:168)
	at com.intellij.util.indexing.UpdateTask.process(UpdateTask.java:63)
	at com.intellij.util.indexing.UpdateTask.processAll(UpdateTask.java:32)
	at com.intellij.util.indexing.FileBasedIndexImpl.lambda$indexUnsavedDocuments$13(FileBasedIndexImpl.java:1011)
	at com.intellij.util.indexing.StorageBufferingHandler.runUpdate(StorageBufferingHandler.java:20)
	at com.intellij.util.indexing.FileBasedIndexImpl.indexUnsavedDocuments(FileBasedIndexImpl.java:1011)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:795)
	at com.intellij.util.indexing.FileBasedIndexEx.processExceptions(FileBasedIndexEx.java:244)
	at com.intellij.util.indexing.FileBasedIndexEx.processValueIterator(FileBasedIndexEx.java:349)
	at com.intellij.util.indexing.FileBasedIndexEx.processValuesInScope(FileBasedIndexEx.java:318)
	at com.intellij.util.indexing.FileBasedIndexEx.getValues(FileBasedIndexEx.java:111)
	at com.magento.idea.magento2plugin.indexes.JsIndex.getRequireJsPreferences(JsIndex.java:46)
	at com.magento.idea.magento2plugin.reference.provider.RequireJsPreferenceReferenceProvider.getReferencesByElement(RequireJsPreferenceReferenceProvider.java:20)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.getReferences(ReferenceProvidersRegistryImpl.java:199)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.mapNotEmptyReferencesFromProviders(ReferenceProvidersRegistryImpl.java:162)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.doGetReferencesFromProviders(ReferenceProvidersRegistryImpl.java:141)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry.lambda$getReferencesFromProviders$0(ReferenceProvidersRegistry.java:39)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:158)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:39)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$3(CachedValueBase.java:227)
	at com.intellij.util.CachedValueBase.computeData(CachedValueBase.java:42)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$4(CachedValueBase.java:227)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:43)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:228)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:28)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:72)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:155)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:121)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry.getReferencesFromProviders(ReferenceProvidersRegistry.java:38)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry.getReferencesFromProviders(ReferenceProvidersRegistry.java:31)
	at com.intellij.lang.javascript.psi.impl.JSLiteralExpressionImpl.createRefs(JSLiteralExpressionImpl.java:103)
	at com.intellij.lang.javascript.psi.impl.JSLiteralExpressionImpl.lambda$static$0(JSLiteralExpressionImpl.java:78)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.doCompute(PsiParameterizedCachedValue.java:46)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$3(CachedValueBase.java:227)
	at com.intellij.util.CachedValueBase.computeData(CachedValueBase.java:42)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$4(CachedValueBase.java:227)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:43)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:228)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.getValue(PsiParameterizedCachedValue.java:35)
	at com.intellij.psi.util.CachedValuesManager.getParameterizedCachedValue(CachedValuesManager.java:82)
	at com.intellij.lang.javascript.psi.impl.JSLiteralExpressionImpl.getReferences(JSLiteralExpressionImpl.java:88)
	at com.intellij.lang.javascript.validation.JSAnnotatingVisitor.checkReferences(JSAnnotatingVisitor.java:268)
	at com.intellij.lang.javascript.validation.JSAnnotatingVisitor.checkReferences(JSAnnotatingVisitor.java:259)
	at com.intellij.lang.javascript.validation.JSAnnotatingVisitor.visitJSLiteralExpression(JSAnnotatingVisitor.java:194)
	at com.intellij.lang.javascript.psi.impl.JSLiteralExpressionImpl.accept(JSLiteralExpressionImpl.java:128)
	at com.intellij.lang.javascript.validation.JSAnnotator.annotate(JSAnnotator.java:17)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.runAnnotators(DefaultHighlightVisitor.java:134)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:114)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.runVisitors(GeneralHighlightingPass.java:335)
	at com.intellij.codeInsight.daemon.impl.Gene
```

**To Reproduce** (*)

1. Right click vendor namespace and select New -> Magento 2 Module
2. Fill our required module information in click Ok to create the module
3. Module is created but the dark overlay remains

**Expected behavior** (*)

Overlay should be cleared after creating a new module.

**Please complete the following information:** (*)

 - OS: Mac OS X 11.6
 - PhpStorm/Intellij version: 2021.2.2
 - Plugin Version: 4.1.0

**Comment by msyhr** [comment]
@mts527 Do you happen to be using the material theme UI plugin  as well?

I was having this issue (with every right click > new Magento module/module file) after their most recent update to the plugin.

If I disable the Magento plugin: bug goes away.

If I disable the UI theme: bug goes away.

Both work as expected separately.

I was able to fix this by going into the Material UI settings and unchecking the "Show Overlays" option

![image](https://user-images.githubusercontent.com/22968596/136089026-ccc4d263-9a0c-412a-bf0a-bef01ddaf84d.png)

**Comment by mts527** [comment]
@msyhr Yes, I'm using Material UI plugin and unchecking "Show Overlays" did solve the issue, thanks! :)

**Comment by bohdan-harniuk** [comment]
Hello, @mts527!

Thank you for your report!
We'll try to deliver fix in the next bug-fix release!
To be able to use both plugins if it can be achieved by fixing our codebase.

Thank you!

**Comment by bohdan-harniuk** [comment]
Hello, @mts527, @msyhr!

Thank you for the detailed description of the bug and steps to reproduce!
It helped a lot! 

The bug is fixed and it will be delivered in the next bug-fix release (version 4.2.1).
 
Thank you!
